errors.go

```diff
@@ -42,10 +42,25 @@ type ErrorObject struct {
 	// Code is an application-specific error code, expressed as a string value.
 	Code string `json:"code,omitempty"`
 
+	// Source is an object containing references to the primary source of the error.
+	Source *ErrorSource `json:"source,omitempty"`
+
 	// Meta is an object containing non-standard meta-information about the error.
 	Meta *map[string]interface{} `json:"meta,omitempty"`
 }
 
+// ErrorSource is an object containing references to the primary source of the error.
+type ErrorSource struct {
+	// Pointer is a string indicating the value in the request document that caused the error.
```

Comment:
Possibly mention something about the data format of the string

```suggestion
	// Pointer is a JSON Pointer (RFC6901) indicating the value in the request document that caused the error.
```